### PR TITLE
Add inputFormatters in date picker dialog

### PR DESCRIPTION
**Pull request by Pana-g**
Add inputFormatters in `showDatePicker` function to give users better customizability over the `InputDatePickerFormField`.

Issues: #115151 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above. 
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing. 

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by rubensdemelo** [comment]
any update ?

File: packages/flutter/test/material/date_picker_test.dart [hunk 1]
```diff
@@ -721,6 +724,14 @@ void main() {
       });
     });
 
+    testWidgets('InputFormatters are used', (WidgetTester tester) async {
+      inputFormatters = <TextInputFormatter>[FilteringTextInputFormatter.allow(RegExp('[a-zA-Z]'))];
+      await prepareDatePicker(tester, (Future<DateTime?> date) async {
+        final TextField field = textField(tester);
+        expect(field.inputFormatters?.length, 1);
```
Comment (dnfield) [inline packages/flutter/test/material/date_picker_test.dart:731]: This should either also assert that when inputFormatters is null the length is zero or (better) assert that the input formatters match the list of what was provided.

**Review by justinmc — COMMENTED** [review]
Seems fine to me to pipe inputFormatters through to the TextFormField, though I don't have much context on date pickers.

**Comment by HansMuller** [comment]
Per https://github.com/flutter/flutter/pull/115144#issuecomment-1542640029: 
"an API doc example that demo'd using a custom input formatter would make the use case easier for other developers to understand"

**Comment by HansMuller** [comment]
@Pana-g - are you planning to continue working on this PR per https://github.com/flutter/flutter/pull/115144#issuecomment-1632845105 ?

**Comment by Hixie** [comment]
@Pana-g thanks for your contributions on this! I've added a link to this PR to the issue so that the next person to start working on this can start with your work. I'm going to close this PR for now. Thanks!

**Pull request closed without merging**